src/MakersOfDenmark.Application/Commands/V1/admin/AddMakerSpaceTool.cs

```diff
@@ -0,0 +1,43 @@
+﻿using MakersOfDenmark.Domain.Models;
+using MakersOfDenmark.Infrastructure.Persistence;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MakersOfDenmark.Application.Commands.V1.admin
+{
+    public class AddMakerSpaceTool : IRequest
+    {
+        public Guid Id { get; set; }
+        public string Make { get; set; }
+        public string Model { get; set; }
+    }
+    public class AddMakerSpaceToolsHandler : IRequestHandler<AddMakerSpaceTool>
+    {
+        private readonly MODContext _context;
+
+        public AddMakerSpaceToolsHandler(MODContext context)
+        {
+            _context = context;
+        }
+        public async Task<Unit> Handle(AddMakerSpaceTool request, CancellationToken cancellationToken = default)
+        {
+            var tool = new Tool { Make = request.Make, Model = request.Model };
+            var makerSpace = await _context.MakerSpace.Include(x => x.Tools).FirstOrDefaultAsync(x => x.Id == request.Id);
+            var msTool = makerSpace.Tools.FirstOrDefault(x => x.Make == request.Make && x.Model == request.Model);
+            if (msTool != default)
+            {
+                throw new Exception("Tool already exists on MakerSpace");
```

Comment:
Jeg ville ikke bruge en Exception her. Jeg tænker Fluentvalidation skal bruges til det her i stedet, men en Customvalidation der tjekker om den eksisterer. 

Exceptions er kun for exceptionelle cases. Et tool der allerede eksisterer kan vi godt komme ud af.

Reply:
Skal jeg bare ignorere forsøget på at tilføje den så? 
`return new Unit();`

Reply:
Det kunne være en mulighed - hvad returnerer Unit?

Reply:
Unit er en måde at sige NULL på... Dette er ikke længere relevant